flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java

```diff
@@ -49,6 +64,22 @@ public EventBus reactor(MDCCleanerThreadPoolExecutor threadPoolExecutor, Environ
                 .env(env)
                 .dispatcher(new ThreadPoolExecutorDispatcher(eventBusThreadPoolBacklogSize, eventBusThreadPoolCoreSize, threadPoolExecutor))
                 .traceEventPath()
+                .dispatchErrorHandler(throwable -> {
+                    if (throwable.getCause() instanceof Exceptions.ValueCause) {
+                        if (((Exceptions.ValueCause) throwable.getCause()).getValue() instanceof Event) {
+                            Event event = (Event) ((Exceptions.ValueCause) throwable.getCause()).getValue();
+                            if (event.getHeaders() != null && event.getHeaders().get("FLOW_ID") != null) {
+                                String flowId = event.getHeaders().get("FLOW_ID").toString();
+                                LOGGER.error("Unhandled exception happened in flow {}, lets cancel it", flowId, throwable);
```

Comment:
would it make sense to add all the headers to the log line?

Reply:
Makes sense, I added some log line which can be helpful.